### PR TITLE
chore: replace proxy mining with sha mining in tests

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Replaced merge mining proxy mining in most tests with sha3 miner. 
Added base_node state to grpc for use in the tests

Motivation and Context
---
The monero merge mining proxy makes a network call to the outside which can be unreliable. The sha3 miner only uses local information which is faster as well. 

We use sleeps that in the cucumber code to ensure that base_nodes have time to sync, we can just if they are in listening state to check if we can go to the next step

How Has This Been Tested?
---
Manual cucumber tests

### CODE REVIEW

File: integration_tests/features/Mempool.feature [hunk 1]
```diff
@@ -96,7 +96,7 @@ Feature: Mempool
     Then SENDER has TX1 in NOT_STORED state
     Then SENDER has TX2 in MINED state
 
-  @critical
+  @critical @flaky
```
Comment (stringhandler) [inline integration_tests/features/Mempool.feature:99]: How is this regressing now?

**Review by stringhandler — COMMENTED** [review]
I'm not keen on this. It makes the code more complicated instead of simpler. 

There are a number of tests that are now marked as flaky, an indication that this is a regression

File: integration_tests/features/Mempool.feature (hunk 1 above)
Comment (SWvheerden) [reply integration_tests/features/Mempool.feature:99]: Its a new test, created in PR #3439 
It failed on CI this morning and I have not been able to get it to fail locally.

**Review by philipr-za — COMMENTED** [review]
This is looking good. I like the addition of waiting on the base node state. One comment and we will tackle Flakey tests in the next sprint.

File: integration_tests/features/Reorgs.feature [hunk 2]
```diff
@@ -197,6 +199,8 @@ Feature: Reorgs
     And I connect node NODE_B1 to node NODE_B3 and wait 1 seconds
     And I connect node NODE_B2 to node NODE_B4 and wait 1 seconds
     And I connect node SEED_B1 to node SEED_B2 and wait <SYNC_TIME> seconds
```
Comment (stringhandler) [inline integration_tests/features/Reorgs.feature:201]: Perhaps we can remove this `wait <SYNC_TIME> seconds` and then keep the below

**Review by stringhandler — APPROVED** [review]
Ok, looking at this again I see that the state check generally replaces a wait X seconds. I think this can go in.

**Pull request merged**